From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GGreenix <76048260+GGreenix@users.noreply.github.com>
Date: Thu, 2 Jan 2025 00:05:47 +0200
Subject: [PATCH 11/17] change names of roi variables

---
 apriltag.c | 4 ++--
 apriltag.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/apriltag.c b/apriltag.c
index be07cdb566ef6b0811faf1e318421dca0bfb51fb..e08b0f9cbabe325380f5ff64eca9fa09788cd18e 100644
--- a/apriltag.c
+++ b/apriltag.c
@@ -1054,8 +1054,8 @@ zarray_t *apriltag_detector_detect(apriltag_detector_t *td, image_u8_t *im_orig)
                 image_u8_t *orig = image_u8_copy(quad_im);
                 image_u8_gaussian_blur(quad_im, sigma, ksz);
 
-                for (int y = td->roiYCoor; y < td->roiHeight; y++) {
-                    for (int x = td->roiXCoor; x < td->roiWidth; x++) {
+                for (int y = td->roiY; y < td->roiHeight; y++) {
+                    for (int x = td->roiX; x < td->roiWidth; x++) {
                         int vorig = orig->buf[y*orig->stride + x];
                         int vblur = quad_im->buf[y*quad_im->stride + x];
 
diff --git a/apriltag.h b/apriltag.h
index 3704489c897186b597a87776a0cd1f5bd7a33541..0be34c4d72939bf8aaf93d09f7f21f06c6783315 100644
--- a/apriltag.h
+++ b/apriltag.h
@@ -190,7 +190,7 @@ struct apriltag_detector
     pthread_mutex_t mutex;
 
     //ROI used for saving proccesing time
-    double roiXCoor, roiYCoor, roiWidth, roiHeight;
+    int roiX, roiY, roiWidth, roiHeight;
 };
 
 // Represents the detection of a tag. These are returned to the user
