From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GGreenix <76048260+GGreenix@users.noreply.github.com>
Date: Tue, 7 Jan 2025 16:57:28 +0200
Subject: [PATCH 16/17] fixed roiX fatal error

---
 apriltag_quad_thresh.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/apriltag_quad_thresh.c b/apriltag_quad_thresh.c
index 454258bc936f3ffa5ff5519520dc6f3367e6b1c4..4d4ae24b0ff4ce7eb69accd92fbd91d79cdd4d1c 100644
--- a/apriltag_quad_thresh.c
+++ b/apriltag_quad_thresh.c
@@ -1075,13 +1075,11 @@ static void do_quad_task(void *p)
 
         struct quad quad;
         memset(&quad, 0, sizeof(struct quad));
-        // if(&quad->p[3][0] > td->roiX && &quad->p[3][1] > td->roiY && &quad->p[1][0] < td->roiX + td->roiWidth && &quad->p[1][1] < td->roiY + td->roiHeight){    
-            if (fit_quad(td, task->im, *cluster, &quad, task->tag_width, task->normal_border, task->reversed_border)) {
-                pthread_mutex_lock(&td->mutex);
-                zarray_add(quads, &quad);
-                pthread_mutex_unlock(&td->mutex);
-            }
-        // }
+        if (fit_quad(td, task->im, *cluster, &quad, task->tag_width, task->normal_border, task->reversed_border)) {
+            pthread_mutex_lock(&td->mutex);
+            zarray_add(quads, &quad);
+            pthread_mutex_unlock(&td->mutex);
+        }
     }
 }
 
