--- a/maven_deps.bzl
+++ b/maven_deps.bzl
@@ -14,6 +14,19 @@ def __setup_jdk_dependencies(mctx):
         strip_prefix = "jdk-17.0.12+7",
     )
 
+    remote_java_repository(
+        name = "roboriojdk_linux_arm32",
+        prefix = "roboriojdk",
+        version = "17",
+        target_compatible_with = [
+            "@platforms//os:linux",
+            "@platforms//cpu:armv7",
+        ],
+        sha256 = "ce7873ebf40ed0eb1089941ead4d3af79a205b1264f3162860d26ae957572b74",
+        urls = ["https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.12%2B7/OpenJDK17U-jdk_arm_linux_hotspot_17.0.12_7.tar.gz"],
+        strip_prefix = "jdk-17.0.12+7",
+    )
+
     remote_java_repository(
         name = "roboriojdk_linux_arm64",
         prefix = "roboriojdk",
@@ -19,8 +19,8 @@
         version = "17",
         target_compatible_with = [
             "@platforms//os:linux",
-            "@platforms//cpu:armv7",
+            "@platforms//cpu:aarch64",
         ],
         sha256 = "8257de06bf37f0c8f19f8d542e2ab5a4e17db3ca5f29d041bd0b02ab265db021",
         urls = ["https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.12%2B7/OpenJDK17U-jdk_aarch64_linux_hotspot_17.0.12_7.tar.gz"],
         strip_prefix = "jdk-17.0.12+7",
@@ -71,6 +84,7 @@ def setup_legacy_setup_jdk_dependencies():
 
     REMOTE_JDK_REPOS = [
         "roboriojdk_linux",
+        "roboriojdk_linux_arm32",
         "roboriojdk_linux_arm64",
         "roboriojdk_mac",
         "roboriojdk_mac_arm64",
