diff --git a/wpilibc/src/test/native/cpp/UnitNetworkTablesTest.cpp b/wpilibc/src/test/native/cpp/UnitNetworkTablesTest.cpp
index ee315fc464..0576be9b2c 100644
--- a/wpilibc/src/test/native/cpp/UnitNetworkTablesTest.cpp
+++ b/wpilibc/src/test/native/cpp/UnitNetworkTablesTest.cpp
@@ -23,7 +23,7 @@ TEST_F(UnitNetworkTablesTest, Publish) {
       wpi::nt::UnitTopic<wpi::units::meters<>>{inst.GetTopic("meterTest")};
   auto pub = topic.Publish();
   pub.Set(2_m);
-  ASSERT_EQ(topic.GetProperty("unit"), "meter");
+  ASSERT_EQ(topic.GetProperty("unit"), "meters");
   ASSERT_TRUE(topic.IsMatchingUnit());
 }
 
diff --git a/wpimath/robotpy_native_build_info.bzl b/wpimath/robotpy_native_build_info.bzl
index 103ac9ee62..16ab09f0f8 100644
--- a/wpimath/robotpy_native_build_info.bzl
+++ b/wpimath/robotpy_native_build_info.bzl
@@ -9,6 +9,7 @@ def define_native_wrapper(name, pyproject_toml = None):
         srcs = native.glob(["src/main/native/include/**"]) + native.glob(["src/generated/main/native/include/**"], allow_empty = True) + native.glob([
             "src/main/native/thirdparty/gcem/include/**",
             "src/main/native/thirdparty/sleipnir/include/**",
+            "src/main/native/thirdparty/units/include/**",
         ]),
         out = "native/wpimath/include",
         root_paths = ["src/main/native/include/"],
@@ -17,6 +18,7 @@ def define_native_wrapper(name, pyproject_toml = None):
             "wpimath/src/main/native/include": "",
             "wpimath/src/main/native/thirdparty/gcem/include": "",
             "wpimath/src/main/native/thirdparty/sleipnir/include": "",
+            "wpimath/src/main/native/thirdparty/units/include": "",
         },
         verbose = False,
         visibility = ["//visibility:public"],
diff --git a/wpimath/src/main/native/include/wpi/math/controller/ElevatorFeedforward.hpp b/wpimath/src/main/native/include/wpi/math/controller/ElevatorFeedforward.hpp
index ef88d60c4c..2a85140058 100644
--- a/wpimath/src/main/native/include/wpi/math/controller/ElevatorFeedforward.hpp
+++ b/wpimath/src/main/native/include/wpi/math/controller/ElevatorFeedforward.hpp
@@ -7,8 +7,10 @@
 #include "wpi/math/controller/LinearPlantInversionFeedforward.hpp"
 #include "wpi/math/linalg/EigenCore.hpp"
 #include "wpi/math/util/MathShared.hpp"
+#include <wpi/units/acceleration.h>
 #include <wpi/units/length.h>
 #include <wpi/units/time.h>
+#include <wpi/units/velocity.h>
 #include <wpi/units/voltage.h>
 #include "wpi/util/MathExtras.hpp"
 
@@ -20,12 +22,8 @@ namespace wpi::math {
 class ElevatorFeedforward {
  public:
   using Distance = wpi::units::meters_;
-  using Velocity =
-      wpi::units::compound_conversion_factor<Distance,
-                                wpi::units::inverse<wpi::units::seconds_>>;
-  using Acceleration =
-      wpi::units::compound_conversion_factor<Velocity,
-                                wpi::units::inverse<wpi::units::seconds_>>;
+  using Velocity = wpi::units::meters_per_second_;
+  using Acceleration = wpi::units::meters_per_second_squared_;
   using kv_unit = wpi::units::compound_conversion_factor<wpi::units::volts_,
                                             wpi::units::inverse<Velocity>>;
   using ka_unit = wpi::units::compound_conversion_factor<wpi::units::volts_,
diff --git a/wpimath/src/main/native/include/wpi/math/kinematics/DifferentialDriveKinematics.hpp b/wpimath/src/main/native/include/wpi/math/kinematics/DifferentialDriveKinematics.hpp
index 466f3911a3..f75419ec9e 100644
--- a/wpimath/src/main/native/include/wpi/math/kinematics/DifferentialDriveKinematics.hpp
+++ b/wpimath/src/main/native/include/wpi/math/kinematics/DifferentialDriveKinematics.hpp
@@ -15,6 +15,7 @@
 #include "wpi/math/kinematics/Kinematics.hpp"
 #include "wpi/math/util/MathShared.hpp"
 #include <wpi/units/angle.h>
+#include <wpi/units/frequency.h>
 #include <wpi/units/length.h>
 #include "wpi/util/SymbolExports.hpp"
 
diff --git a/wpimath/src/main/native/include/wpi/math/trajectory/ExponentialProfile.hpp b/wpimath/src/main/native/include/wpi/math/trajectory/ExponentialProfile.hpp
index 5d58675230..98ed99dd30 100644
--- a/wpimath/src/main/native/include/wpi/math/trajectory/ExponentialProfile.hpp
+++ b/wpimath/src/main/native/include/wpi/math/trajectory/ExponentialProfile.hpp
@@ -4,6 +4,7 @@
 
 #pragma once
 
+#include <wpi/units/frequency.h>
 #include <wpi/units/time.h>
 
 namespace wpi::math {
@@ -48,7 +49,7 @@ class ExponentialProfile {
       wpi::units::compound_conversion_factor<Velocity,
                                 wpi::units::inverse<wpi::units::seconds_>>;
   using Input_t = wpi::units::unit<Input>;
-  using A_t = wpi::units::unit<wpi::units::inverse<wpi::units::seconds_>>;
+  using A_t = wpi::units::hertz<>;
   using B_t = wpi::units::unit<
       wpi::units::compound_conversion_factor<Acceleration, wpi::units::inverse<Input>>>;
   using KV = wpi::units::compound_conversion_factor<Input, wpi::units::inverse<Velocity>>;
diff --git a/wpimath/src/main/python/pyproject.toml b/wpimath/src/main/python/pyproject.toml
index 8b98276897..e0b4f12df5 100644
--- a/wpimath/src/main/python/pyproject.toml
+++ b/wpimath/src/main/python/pyproject.toml
@@ -1453,7 +1453,7 @@ default_arg_cast = true
 [[tool.semiwrap.export_type_casters.wpimath-casters.headers]]
 header = "units_compound_type_caster.h"
 types = [
-    "wpi::units::compound_unit",
+    "wpi::units::compound_conversion_factor",
     "wpi::units::inverse",
 ]
 default_arg_cast = false
@@ -1461,20 +1461,14 @@ default_arg_cast = false
 [[tool.semiwrap.export_type_casters.wpimath-casters.headers]]
 header = "units_misc_type_caster.h"
 types = [
-    "wpi::units::dimensionless_t",
-    "wpi::units::dimensionless::dimensionless_t",
-    "wpi::units::scalar_t",
-    "wpi::units::dimensionless::scalar_t",
+    "wpi::units::dimensionless",
 ]
 default_arg_cast = true
 
 [[tool.semiwrap.export_type_casters.wpimath-casters.headers]]
 header = "units_misc_type_caster.h"
 types = [
-    "wpi::units::dimensionless",
-    "wpi::units::dimensionless::dimensionless",
-    "wpi::units::scalar",
-    "wpi::units::dimensionless::scalar",
+    "wpi::units::dimensionless_",
 ]
 default_arg_cast = false
 
diff --git a/wpimath/src/main/python/semiwrap/LinearQuadraticRegulator.yml b/wpimath/src/main/python/semiwrap/LinearQuadraticRegulator.yml
index f9e2cede17..3c86358d23 100644
--- a/wpimath/src/main/python/semiwrap/LinearQuadraticRegulator.yml
+++ b/wpimath/src/main/python/semiwrap/LinearQuadraticRegulator.yml
@@ -8,7 +8,8 @@ classes:
         overloads:
           const LinearSystem<States, Inputs, Outputs>&, const StateArray&, const InputArray&, wpi::units::seconds<>:
             ignore: true
-          const Matrixd<States, States>&, const Matrixd<States, Inputs>&, const StateArray&, const InputArray&, wpi::units::seconds<>:
+          ? const Matrixd<States, States>&, const Matrixd<States, Inputs>&, const StateArray&, const InputArray&, wpi::units::seconds<>
+          :
           ? const Matrixd<States, States>&, const Matrixd<States, Inputs>&, const Matrixd<States, States>&, const Matrixd<Inputs, Inputs>&, wpi::units::seconds<>
           :
           ? const Matrixd<States, States>&, const Matrixd<States, Inputs>&, const Matrixd<States, States>&, const Matrixd<Inputs, Inputs>&, const Matrixd<States, Inputs>&, wpi::units::seconds<>
diff --git a/wpimath/src/main/python/wpimath/_impl/src/type_casters/_units_base_type_caster.h b/wpimath/src/main/python/wpimath/_impl/src/type_casters/_units_base_type_caster.h
index 9566128b1b..3481866255 100644
--- a/wpimath/src/main/python/wpimath/_impl/src/type_casters/_units_base_type_caster.h
+++ b/wpimath/src/main/python/wpimath/_impl/src/type_casters/_units_base_type_caster.h
@@ -16,19 +16,19 @@ namespace detail {
   When going from Python to C++, the units a user uses are once again
   whatever the C++ function specifies:
 
-    void setFeet(wpi::units::feet<> ft);   // must pass a float, it's in feet
+    void setFeet(wpi::units::feet<> ft);    // must pass a float, it's in feet
     void setMeters(wpi::units::meters<> m); // must pass a float, it's in meters
 
   Unfortunately, with this type caster and robotpy-build there are mismatch
   issues with implicit conversions when default values are used that don't
   match the actual value:
 
-    foo(wpi::units::seconds<> tm = 10_ms);    // if not careful, pybind11 will 
-                                        // store as 10 seconds
+    foo(wpi::units::seconds<> tm = 10_ms);    // if not careful, pybind11 will
+                                              // store as 10 seconds
 */
-template <class U, typename T, template <typename> class S>
-struct type_caster<wpi::units::unit<U, T, S>> {
-  using value_type = wpi::units::unit<U, T, S>;
+template <class CF, typename T, typename S>
+struct type_caster<wpi::units::unit<CF, T, S>> {
+  using value_type = wpi::units::unit<CF, T, S>;
 
   // TODO: there should be a way to include the type with this
   PYBIND11_TYPE_CASTER(value_type, handle_type_name<value_type>::name);
@@ -52,4 +52,4 @@ struct type_caster<wpi::units::unit<U, T, S>> {
 };
 
 } // namespace detail
-} // namespace pybind11
\ No newline at end of file
+} // namespace pybind11
diff --git a/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_compound_type_caster.h b/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_compound_type_caster.h
index e154cd924e..4da3f4d98b 100644
--- a/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_compound_type_caster.h
+++ b/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_compound_type_caster.h
@@ -33,54 +33,54 @@ struct handle_type_name<
   static constexpr auto name = _("wpimath.units.units_per_second_squared");
 };
 
-using volt_seconds = wpi::units::compound_conversion_factor<wpi::units::volts_, wpi::units::seconds_>;
-using volt_seconds_squared = wpi::units::compound_conversion_factor<volt_seconds, wpi::units::seconds_>;
+using volt_seconds_ = wpi::units::compound_conversion_factor<wpi::units::volts_, wpi::units::seconds_>;
+using volt_seconds_squared_ = wpi::units::compound_conversion_factor<volt_seconds_, wpi::units::seconds_>;
 
-template <> struct handle_type_name<wpi::units::unit<volt_seconds>> {
+template <> struct handle_type_name<wpi::units::unit<volt_seconds_>> {
   static constexpr auto name = _("wpimath.units.volt_seconds");
 };
 
-template <> struct handle_type_name<wpi::units::unit<volt_seconds_squared>> {
+template <> struct handle_type_name<wpi::units::unit<volt_seconds_squared_>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_squared");
 };
 
 template <>
 struct handle_type_name<wpi::units::unit<
-    wpi::units::compound_conversion_factor<volt_seconds, wpi::units::inverse<wpi::units::meters_>>>> {
+    wpi::units::compound_conversion_factor<volt_seconds_, wpi::units::inverse<wpi::units::meters_>>>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_per_meter");
 };
 template <>
 struct handle_type_name<wpi::units::unit<
-    wpi::units::compound_conversion_factor<volt_seconds_squared, wpi::units::inverse<wpi::units::meters_>>>> {
+    wpi::units::compound_conversion_factor<volt_seconds_squared_, wpi::units::inverse<wpi::units::meters_>>>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_squared_per_meter");
 };
 template <>
 struct handle_type_name<wpi::units::unit<
-    wpi::units::compound_conversion_factor<volt_seconds, wpi::units::inverse<wpi::units::feet_>>>> {
+    wpi::units::compound_conversion_factor<volt_seconds_, wpi::units::inverse<wpi::units::feet_>>>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_per_feet");
 };
 template <>
 struct handle_type_name<wpi::units::unit<
-    wpi::units::compound_conversion_factor<volt_seconds_squared, wpi::units::inverse<wpi::units::feet_>>>> {
+    wpi::units::compound_conversion_factor<volt_seconds_squared_, wpi::units::inverse<wpi::units::feet_>>>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_squared_per_feet");
 };
 template <>
 struct handle_type_name<wpi::units::unit<
-    wpi::units::compound_conversion_factor<volt_seconds, wpi::units::inverse<wpi::units::radians_>>>> {
+    wpi::units::compound_conversion_factor<volt_seconds_, wpi::units::inverse<wpi::units::radians_>>>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_per_radian");
 };
 template <>
 struct handle_type_name<wpi::units::unit<wpi::units::compound_conversion_factor<
-    volt_seconds_squared, wpi::units::inverse<wpi::units::radians_>>>> {
+    volt_seconds_squared_, wpi::units::inverse<wpi::units::radians_>>>> {
   static constexpr auto name = _("wpimath.units.volt_seconds_squared_per_radian");
 };
 
-using unit_seconds = wpi::units::compound_conversion_factor<wpi::units::dimensionless_, wpi::units::seconds_>;
-using unit_seconds_squared = wpi::units::compound_conversion_factor<unit_seconds, wpi::units::seconds_>;
+using unit_seconds_ = wpi::units::compound_conversion_factor<wpi::units::dimensionless_, wpi::units::seconds_>;
+using unit_seconds_squared_ = wpi::units::compound_conversion_factor<unit_seconds_, wpi::units::seconds_>;
 
 template <>
 struct handle_type_name<wpi::units::unit<
-    wpi::units::compound_conversion_factor<unit_seconds_squared, wpi::units::inverse<wpi::units::dimensionless_>>>> {
+    wpi::units::compound_conversion_factor<unit_seconds_squared_, wpi::units::inverse<wpi::units::dimensionless_>>>> {
   static constexpr auto name = _("wpimath.units.unit_seconds_squared_per_unit");
 };
 
diff --git a/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_misc_type_caster.h b/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_misc_type_caster.h
index 511df827d2..dafe4a999a 100644
--- a/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_misc_type_caster.h
+++ b/wpimath/src/main/python/wpimath/_impl/src/type_casters/units_misc_type_caster.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <wpi/units/mass.h>
+#include <wpi/units/core.h>
 
 namespace pybind11 {
 namespace detail {
@@ -13,15 +13,7 @@ template <> struct handle_type_name<wpi::units::dimensionless_> {
   static constexpr auto name = _("float");
 };
 
-// template <> struct handle_type_name<wpi::units::dimensionless<>> {
-//   static constexpr auto name = _("float");
-// };
-
-// template <> struct handle_type_name<wpi::units::dimensionless_> {
-//   static constexpr auto name = _("float");
-// };
-
 } // namespace detail
 } // namespace pybind11
 
-#include "_units_base_type_caster.h"
\ No newline at end of file
+#include "_units_base_type_caster.h"
