From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jade Turner <spacey-sooty@proton.me>
Date: Fri, 11 Apr 2025 11:20:41 +0800
Subject: [PATCH] Set IMGUI_DEFINE_MATH_OPERATORS

Signed-off-by: Jade Turner <spacey-sooty@proton.me>
---
 imconfig.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imconfig.h b/imconfig.h
index b56ba49467ebb388e038f0a23aed46044085bdff..291ce1995412f10e784beaf3f4c51e02d90b9813 100644
--- a/imconfig.h
+++ b/imconfig.h
@@ -98,7 +98,7 @@
         operator MyVec4() const { return MyVec4(x,y,z,w); }
 */
 //---- ...Or use Dear ImGui's own very basic math operators.
-//#define IMGUI_DEFINE_MATH_OPERATORS
+#define IMGUI_DEFINE_MATH_OPERATORS
 
 //---- Use 32-bit vertex indices (default is 16-bit) is one way to allow large meshes with more than 64K vertices.
 // Your renderer backend will need to support it (most example renderer backends support both 16/32-bit indices).
