From 492ec25f2627f24fa1b57f0a060ddc530310d5a4 Mon Sep 17 00:00:00 2001
From: PJ Reiniger <pj.reiniger@gmail.com>
Date: Mon, 9 May 2022 00:18:29 -0400
Subject: [PATCH 23/29] Add convienence feature to SmallString

---
 llvm/include/llvm/ADT/SmallString.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/llvm/include/llvm/ADT/SmallString.h b/llvm/include/llvm/ADT/SmallString.h
index 7bf0f5589..020022fd5 100644
--- a/llvm/include/llvm/ADT/SmallString.h
+++ b/llvm/include/llvm/ADT/SmallString.h
@@ -189,9 +189,11 @@ public:
   /// Implicit conversion to std::string_view.
   operator std::string_view() const { return str(); }
 
-  explicit operator std::string() const {
-    return std::string(this->data(), this->size());
-  }
+  /// Explicit conversion to std::string.
+  std::string string() const { return {this->begin(), this->size()}; }
+
+  /// Implicit conversion to std::string.
+  operator std::string() const { return string(); }
 
   // Extra operators.
   SmallString &operator=(std::string_view RHS) {
-- 
2.37.2

