From 498e56e09e8f264978831519ff9c8afa701bf208 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Wed, 10 Aug 2022 22:35:00 -0700
Subject: [PATCH 28/28] Remove StringMap test for llvm::sort()

---
 llvm/unittests/ADT/StringMapTest.cpp | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/llvm/unittests/ADT/StringMapTest.cpp b/llvm/unittests/ADT/StringMapTest.cpp
index de6daf3da..ca41631cc 100644
--- a/llvm/unittests/ADT/StringMapTest.cpp
+++ b/llvm/unittests/ADT/StringMapTest.cpp
@@ -308,20 +308,6 @@ TEST_F(StringMapTest, InsertOrAssignTest) {
   EXPECT_EQ(0, try1.first->second.copy);
 }
 
-TEST_F(StringMapTest, IterMapKeysVector) {
-  StringMap<int> Map;
-  Map["A"] = 1;
-  Map["B"] = 2;
-  Map["C"] = 3;
-  Map["D"] = 3;
-
-  std::vector<std::string_view> Keys{Map.keys().begin(), Map.keys().end()};
-  llvm::sort(Keys);
-
-  std::vector<std::string_view> Expected{{"A", "B", "C", "D"}};
-  EXPECT_EQ(Expected, Keys);
-}
-
 TEST_F(StringMapTest, IterMapKeysSmallVector) {
   StringMap<int> Map;
   Map["A"] = 1;
