From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Blue <ryanzblue@gmail.com>
Date: Thu, 8 Jun 2023 18:49:52 -0400
Subject: [PATCH 29/29] Fix docs typo in SmallVector

---
 llvm/include/llvm/ADT/SmallVector.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/include/llvm/ADT/SmallVector.h b/llvm/include/llvm/ADT/SmallVector.h
index 4b6bbdeb2ea3e602978733351e352968ccdca36d..84713da54c49125697635381002d81df954915d9 100644
--- a/llvm/include/llvm/ADT/SmallVector.h
+++ b/llvm/include/llvm/ADT/SmallVector.h
@@ -6,7 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 ///
-/// /file
+/// \file
 /// This file defines the SmallVector class.
 ///
 //===----------------------------------------------------------------------===//
