From e723453db031d7e2b1b657120cf9319cd2ba4836 Mon Sep 17 00:00:00 2001
From: PJ Reiniger <pj.reiniger@gmail.com>
Date: Wed, 4 May 2022 18:52:17 -0400
Subject: [PATCH 32/35] Fix documentation warnings

---
 llvm/include/llvm/Support/MemoryBuffer.h | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/llvm/include/llvm/Support/MemoryBuffer.h b/llvm/include/llvm/Support/MemoryBuffer.h
index bdaf98664ff1..cb49838181af 100644
--- a/llvm/include/llvm/Support/MemoryBuffer.h
+++ b/llvm/include/llvm/Support/MemoryBuffer.h
@@ -83,13 +83,6 @@ public:
 
   /// Open the specified file as a MemoryBuffer, returning a new MemoryBuffer
   /// if successful, otherwise returning null.
-  ///
-  /// \param IsText Set to true to indicate that the file should be read in
-  /// text mode.
-  ///
-  /// \param IsVolatile Set to true to indicate that the contents of the file
-  /// can change outside the user's control, e.g. when libclang tries to parse
-  /// while the user is editing/updating the file or if the file is on an NFS.
   static ErrorOr<std::unique_ptr<MemoryBuffer>>
   getFile(const std::string_view &Filename, bool IsText = false,
           bool RequiresNullTerminator = true, bool IsVolatile = false);
@@ -109,10 +102,6 @@ public:
 
   /// Given an already-open file descriptor, read the file and return a
   /// MemoryBuffer.
-  ///
-  /// \param IsVolatile Set to true to indicate that the contents of the file
-  /// can change outside the user's control, e.g. when libclang tries to parse
-  /// while the user is editing/updating the file or if the file is on an NFS.
   static ErrorOr<std::unique_ptr<MemoryBuffer>>
   getOpenFile(sys::fs::file_t FD, const std::string_view &Filename, uint64_t FileSize,
               bool RequiresNullTerminator = true, bool IsVolatile = false);
-- 
2.20.1.windows.1

