From c2ef1fa9bd7687d46131244f031cb5c0fff23155 Mon Sep 17 00:00:00 2001
From: PJ Reiniger <pj.reiniger@gmail.com>
Date: Wed, 4 May 2022 02:59:09 -0400
Subject: [PATCH 33/37] Turn off 16-> 8 conversion check

---
 llvm/lib/Support/ConvertUTFWrapper.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/llvm/lib/Support/ConvertUTFWrapper.cpp b/llvm/lib/Support/ConvertUTFWrapper.cpp
index 893be5c1afd5..543f5f6358ed 100644
--- a/llvm/lib/Support/ConvertUTFWrapper.cpp
+++ b/llvm/lib/Support/ConvertUTFWrapper.cpp
@@ -88,10 +88,6 @@ bool hasUTF16ByteOrderMark(span<const char> S) {
 bool convertUTF16ToUTF8String(span<const UTF16> SrcBytes, SmallVectorImpl<char> &Out) {
   assert(Out.empty());
 
-  // Error out on an uneven byte count.
-  if (SrcBytes.size() % 2)
-    return false;
-
   // Avoid OOB by returning early on empty input.
   if (SrcBytes.empty())
     return true;
-- 
2.20.1.windows.1

