From 056b1f2f3fafe2fa4555b35451189aa771c3d6cc Mon Sep 17 00:00:00 2001
From: PJ Reiniger <pj.reiniger@gmail.com>
Date: Wed, 4 May 2022 19:00:55 -0400
Subject: [PATCH 33/35] Fix athena compilation error

---
 llvm/lib/Support/MemoryBuffer.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/llvm/lib/Support/MemoryBuffer.cpp b/llvm/lib/Support/MemoryBuffer.cpp
index 949c459250c1..c44d13b05cd2 100644
--- a/llvm/lib/Support/MemoryBuffer.cpp
+++ b/llvm/lib/Support/MemoryBuffer.cpp
@@ -159,7 +159,7 @@ getMemBufferCopyImpl(span<const uint8_t> InputData, std::string_view BufferName)
   if (!Buf)
     return make_error_code(errc::not_enough_memory);
   memcpy(Buf->getBufferStart(), InputData.data(), InputData.size());
-  return std::move(Buf);
+  return ErrorOr<std::unique_ptr<WritableMemoryBuffer>>(std::move(Buf));
 }
 
 std::unique_ptr<MemoryBuffer>
@@ -390,7 +390,7 @@ getReadWriteFile(std::string_view Filename, uint64_t FileSize, uint64_t MapSize,
                                                          Offset, EC));
   if (EC)
     return EC;
-  return std::move(Result);
+  return ErrorOr<std::unique_ptr<WriteThroughMemoryBuffer>>(std::move(Result));
 }
 
 ErrorOr<std::unique_ptr<WriteThroughMemoryBuffer>>
@@ -450,7 +450,7 @@ getOpenFileImpl(sys::fs::file_t FD, std::string_view Filename, uint64_t FileSize
         new (NamedBufferAlloc(Filename)) MemoryBufferMMapFile<MB>(
             RequiresNullTerminator, FD, MapSize, Offset, EC));
     if (!EC)
-      return std::move(Result);
+      return ErrorOr<std::unique_ptr<MB>>(std::move(Result));
   }
 
   auto Buf = WritableMemoryBuffer::getNewUninitMemBuffer(MapSize, Filename);
@@ -489,7 +489,7 @@ getOpenFileImpl(sys::fs::file_t FD, std::string_view Filename, uint64_t FileSize
     BufPtr += NumRead;
   }
 
-  return std::move(Buf);
+  return ErrorOr<std::unique_ptr<MB>>(std::move(Buf));
 }
 
 ErrorOr<std::unique_ptr<MemoryBuffer>>
-- 
2.20.1.windows.1

