From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Tue, 26 May 2026 16:55:28 -0700
Subject: [PATCH 34/34] SmallVector: add cbegin()/cend()/crbegin()/crend()

---
 llvm/include/llvm/ADT/SmallVector.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/llvm/include/llvm/ADT/SmallVector.h b/llvm/include/llvm/ADT/SmallVector.h
index e51771928717625a218981ccd5b9068b63329b02..38d4a23bdd2d46d8b12ccaf27ae44dc6139a54a1 100644
--- a/llvm/include/llvm/ADT/SmallVector.h
+++ b/llvm/include/llvm/ADT/SmallVector.h
@@ -275,14 +275,18 @@ public:
   // forward iterator creation methods.
   iterator begin() { return (iterator)this->BeginX; }
   const_iterator begin() const { return (const_iterator)this->BeginX; }
+  const_iterator cbegin() const { return (const_iterator)this->BeginX; }
   iterator end() { return begin() + size(); }
   const_iterator end() const { return begin() + size(); }
+  const_iterator cend() const { return begin() + size(); }
 
   // reverse iterator creation methods.
   reverse_iterator rbegin()            { return reverse_iterator(end()); }
   const_reverse_iterator rbegin() const{ return const_reverse_iterator(end()); }
+  const_reverse_iterator crbegin() const{ return const_reverse_iterator(end()); }
   reverse_iterator rend()              { return reverse_iterator(begin()); }
   const_reverse_iterator rend() const { return const_reverse_iterator(begin());}
+  const_reverse_iterator crend() const { return const_reverse_iterator(begin());}
 
   size_type size_in_bytes() const { return size() * sizeof(T); }
   size_type max_size() const {
