From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joseph Eng <s-engjo@bsd405.org>
Date: Sun, 28 Apr 2024 19:55:07 -0700
Subject: [PATCH 36/36] Use fmt for gtest formatting of TimePoint

---
 llvm/unittests/Support/Chrono.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/llvm/unittests/Support/Chrono.cpp b/llvm/unittests/Support/Chrono.cpp
index a4d166d435d6d679f773dcf3eab985f0631e12d2..f7c01ec6fc7eefa2d9e40f8ad2f4b4795645693d 100644
--- a/llvm/unittests/Support/Chrono.cpp
+++ b/llvm/unittests/Support/Chrono.cpp
@@ -6,6 +6,10 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <chrono>
+#include <fmt/chrono.h>
+#include <fmt/ostream.h>
+
 #include "llvm/Support/Chrono.h"
 #include "llvm/Support/FormatVariadic.h"
 #include "gtest/gtest.h"
@@ -14,6 +18,15 @@ using namespace llvm;
 using namespace llvm::sys;
 using namespace std::chrono;
 
+namespace std::chrono {
+
+template <typename Clock, typename Duration>
+void PrintTo(const std::chrono::time_point<Clock, Duration>& time_point, std::ostream* os) {
+  fmt::print(*os, "{}", time_point);
+}
+
+}  // namespace std::chrono
+
 namespace {
 
 TEST(Chrono, TimeTConversion) {
