From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joseph Eng <91924258+KangarooKoala@users.noreply.github.com>
Date: Thu, 11 Dec 2025 20:30:41 -0800
Subject: [PATCH 4/9] Group doxygen modules

---
 include/units/core.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/units/core.h b/include/units/core.h
index 57043f35da894a3b1faaf34335996fc2e6a3d7ef..e640b29825b1b319dfec91176eb6015e9c1a08ac 100644
--- a/include/units/core.h
+++ b/include/units/core.h
@@ -414,14 +414,20 @@ namespace units
 	//	DOXYGEN
 	//----------------------------------
 
+	/**
+	 * @defgroup	Units Unit API
+	 */
+
 	/**
 	 * @defgroup	UnitTypes Unit Types
+	 * @ingroup		Units
 	 * @brief		Defines a series of classes which contain dimensioned values. Unit types
 	 *				store a value, and support various arithmetic operations.
 	 */
 
 	/**
 	 * @defgroup	UnitManipulators Unit Manipulators
+	 * @ingroup		Units
 	 * @brief		Defines a series of classes used to manipulate unit types, such as `inverse<>`, `squared<>`, and
 	 *				metric prefixes. Unit manipulators can be chained together, e.g.
 	 *				`inverse<squared<pico<time::seconds>>>` to represent picoseconds^-2.
@@ -429,22 +435,26 @@ namespace units
 
 	/**
 	 * @defgroup	UnitMath Unit Math
+	 * @ingroup		Units
 	 * @brief		Defines a collection of unit-enabled, strongly-typed versions of `<cmath>` functions.
 	 * @details		Includes most c++11 extensions.
 	 */
 
 	/**
 	 * @defgroup	Conversion Explicit Conversion
+	 * @ingroup		Units
 	 * @brief		Functions used to convert values of one logical type to another.
 	 */
 
 	/**
 	 * @defgroup	TypeTraits Type Traits
+	 * @ingroup		Units
 	 * @brief		Defines a series of classes to obtain unit type information at compile-time.
 	 */
 
 	/**
 	 * @defgroup	STDTypeTraits Standard Type Traits Specializations
+	 * @ingroup		Units
 	 * @brief		Specialization of `std::common_type` for unit types.
 	 */
 
