From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gold856 <117957790+Gold856@users.noreply.github.com>
Date: Wed, 22 Jul 2026 18:01:15 -0400
Subject: [PATCH 3/3] Wrap functions in extern "C"

---
 dogleg.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dogleg.h b/dogleg.h
index 7fd4c8df1d04372f389d76ac1fa27b38d21f05f8..933e9e894d9d1e29fb0084c472724f8e19b75112 100644
--- a/dogleg.h
+++ b/dogleg.h
@@ -208,7 +208,9 @@ typedef struct
 
 } dogleg_solverContext_t;
 
-
+#ifdef __cplusplus
+extern "C" {
+#endif
 // Fills in the given structure with the default parameter set
 void dogleg_getDefaultParameters(dogleg_parameters2_t* parameters);
 
@@ -393,3 +395,6 @@ double dogleg_getOutliernessTrace_newFeature_sparse(const double*            Jqu
                                                     int                      NoutlierFeatures,
                                                     dogleg_operatingPoint_t* point,
                                                     dogleg_solverContext_t*  ctx);
+#ifdef __cplusplus
+} // extern "C"
+#endif
