From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gold856 <117957790+Gold856@users.noreply.github.com>
Date: Fri, 22 Nov 2024 09:54:00 -0500
Subject: [PATCH 2/8] Make mrcal functions extern "C"

---
 mrcal.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/mrcal.h b/mrcal.h
index cd600e8f02c66d84ca0210e7c2bb13bf7bf1d1c2..6a0b74d6c1506d2ad39002ebfd195af784843b5b 100644
--- a/mrcal.h
+++ b/mrcal.h
@@ -25,7 +25,9 @@ extern "C" {
 ////////////////////////////////////////////////////////////////////////////////
 //////////////////// Lens models
 ////////////////////////////////////////////////////////////////////////////////
-
+#ifdef __cplusplus
+extern "C" {
+#endif
 // Return an array of strings listing all the available lens models
 //
 // These are all "unconfigured" strings that use "..." placeholders for any
@@ -871,6 +873,9 @@ bool mrcal_read_cameramodel_file_into  (// out
 bool mrcal_write_cameramodel_file(const char* filename,
                                   const mrcal_cameramodel_VOID_t* cameramodel);
 
+#ifdef __cplusplus
+} // extern "C"
+#endif
 #define DECLARE_mrcal_apply_color_map(T,Tname)                          \
     bool mrcal_apply_color_map_##Tname(                                 \
         mrcal_image_bgr_t*    out,                                      \
