From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gold856 <117957790+Gold856@users.noreply.github.com>
Date: Fri, 29 Nov 2024 22:56:22 -0500
Subject: [PATCH 5/8] Use immediately invoked lambda

---
 mrcal.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mrcal.cpp b/mrcal.cpp
index ef5ee3f247cf96839a0a7694be4de6f2a9a47b9d..336e9805bb3a038f437a2a5a59fdb97c451e0fa5 100644
--- a/mrcal.cpp
+++ b/mrcal.cpp
@@ -6668,11 +6668,11 @@ mrcal_optimize( // out
 #if defined ENABLE_TRIANGULATED_WARNINGS && ENABLE_TRIANGULATED_WARNINGS
 #warning "triangulated-solve: this print does not include triangulated outliers"
 #endif
-                 ({MSG("Threw out some outliers. New count = %d/%d (%.1f%%). Going again",
+                 ([=]{MSG("Threw out some outliers. New count = %d/%d (%.1f%%). Going again",
                        stats.Noutliers_board,
                        Nmeasurements_board,
                        (double)(stats.Noutliers_board * 100) / (double)Nmeasurements_board);
-                   true;}));
+                   return true;}()));
 #if defined ENABLE_TRIANGULATED_WARNINGS && ENABLE_TRIANGULATED_WARNINGS
 #warning "triangulated-solve: the above print should deal with triangulated points too"
 #endif
