From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Thu, 16 Jul 2026 21:49:36 -0700
Subject: [PATCH 6/7] Suppress unused-but-set warning

---
 mrcal.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mrcal.cpp b/mrcal.cpp
index e1b2aa78fd6e129dd775caa68545e6afc70a63cb..8ea2c873b28fbdd0d53ccb6af15a6726b1741a68 100644
--- a/mrcal.cpp
+++ b/mrcal.cpp
@@ -4455,7 +4455,7 @@ void optimizer_callback(// input state
 
                        const callback_context_t* ctx)
 {
-    double norm2_error = 0.0;
+    [[maybe_unused]] double norm2_error = 0.0;
 
     int    iJacobian          = 0;
     int    iMeasurement       = 0;
