From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gold856 <117957790+Gold856@users.noreply.github.com>
Date: Wed, 4 Dec 2024 00:58:21 -0500
Subject: [PATCH 7/9] Place all C headers in an extern "C" block

---
 mrcal.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mrcal.cpp b/mrcal.cpp
index 04430c3d0180363d0d2a5a4f83a20f04b842c435..e1c7f2f9dfe5250ded6921e39ae174e7f0a48c36 100644
--- a/mrcal.cpp
+++ b/mrcal.cpp
@@ -14,19 +14,22 @@
 #include <stdlib.h>
 #include <inttypes.h>
 
+extern "C" {
 #include <dogleg.h>
+}
 #include <assert.h>
 #include <stdbool.h>
 #include <math.h>
 #include <string.h>
 
+extern "C" {
 #include "mrcal.h"
 #include "minimath/minimath.h"
 #include "_cahvore.h"
 #include "minimath/minimath-extra.h"
 #include "_util.h"
 #include "scales.h"
-
+}
 #define MSG_IF_VERBOSE(...) do { if(verbose) MSG( __VA_ARGS__ ); } while(0)
 
 #define restrict __restrict
