From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Thu, 16 Jul 2026 21:49:36 -0700
Subject: [PATCH 9/9] Suppress unused-but-set warning

---
 mrcal.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mrcal.cpp b/mrcal.cpp
index 0b7263f58164bfe9f42447741a08366262195f7c..4c8e56c7984ceee7f3708e031ea308102f605747 100644
--- a/mrcal.cpp
+++ b/mrcal.cpp
@@ -4483,7 +4483,7 @@ void optimizer_callback(// input state
 
                        const callback_context_t* ctx)
 {
-    double norm2_error = 0.0;
+    [[maybe_unused]] double norm2_error = 0.0;
 
     int    iJacobian          = 0;
     int    iMeasurement       = 0;
