diff --git a/wpinet/src/main/native/thirdparty/libuv/src/win/fs.cpp b/wpinet/src/main/native/thirdparty/libuv/src/win/fs.cpp
index 26ba53993..7f3a5f328 100644
--- a/wpinet/src/main/native/thirdparty/libuv/src/win/fs.cpp
+++ b/wpinet/src/main/native/thirdparty/libuv/src/win/fs.cpp
@@ -1185,7 +1185,7 @@ void fs__mktemp(uv_fs_t* req, uv__fs_mktemp_func func) {
   size_t len;
   uint64_t v;
   char* path;
-
+
   path = (char*)req->path;
   len = wcslen(req->file.pathw);
   ep = req->file.pathw + len;
@@ -1658,7 +1658,7 @@ INLINE static int fs__stat_handle(HANDLE handle, uv_stat_t* statbuf,
     statbuf->st_mode |= (_S_IREAD | _S_IWRITE) | ((_S_IREAD | _S_IWRITE) >> 3) |
                         ((_S_IREAD | _S_IWRITE) >> 6);
     statbuf->st_nlink = 1;
-    statbuf->st_blksize = 4096;
+    statbuf->st_blksize = 4096;
     statbuf->st_rdev = FILE_DEVICE_NULL << 16;
     return 0;
   }
@@ -1704,7 +1704,7 @@ INLINE static int fs__stat_handle(HANDLE handle, uv_stat_t* statbuf,
    * impossible to delete the file afterwards, since read-only files can't be
    * deleted.
    *
-   * IOW it's all just a clusterfuck and we should think of something that
+   * IOW it's all just a trainwreck and we should think of something that
    * makes slightly more sense.
    *
    * And uv_fs_chmod should probably just fail on windows or be a total no-op.
diff --git a/wpinet/src/main/native/thirdparty/libuv/src/win/tty.cpp b/wpinet/src/main/native/thirdparty/libuv/src/win/tty.cpp
index 7adf3cd3b..c0461ae56 100644
--- a/wpinet/src/main/native/thirdparty/libuv/src/win/tty.cpp
+++ b/wpinet/src/main/native/thirdparty/libuv/src/win/tty.cpp
@@ -1055,7 +1055,7 @@ int uv__tty_read_stop(uv_tty_t* handle) {
     return 0;

   if (handle->flags & UV_HANDLE_TTY_RAW) {
-    /* Cancel raw read. Write some bullshit event to force the console wait to
+    /* Cancel raw read. Write some event to force the console wait to
      * return. */
     memset(&record, 0, sizeof record);
     record.EventType = FOCUS_EVENT;
