From aca70965b0a80000cb8281d097c78a3e3e0feb16 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Tue, 17 May 2022 21:36:57 -0700
Subject: [PATCH 7/7] Squelch GCC 12.1 warnings

---
 src/unix/stream.c | 9 +++++++++
 src/uv-common.c   | 9 +++++++++
 2 files changed, 18 insertions(+)

diff --git a/src/unix/stream.c b/src/unix/stream.c
index f3a8e66d..8aeca159 100644
--- a/src/unix/stream.c
+++ b/src/unix/stream.c
@@ -932,7 +932,16 @@ static void uv__write_callbacks(uv_stream_t* stream) {
   if (QUEUE_EMPTY(&stream->write_completed_queue))
     return;
 
+// FIXME: GCC 12.1 gives a possibly real warning, but we don't know how to fix
+// it
+#if __GNUC__ >= 12
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdangling-pointer="
+#endif  // __GNUC__ >= 12
   QUEUE_MOVE(&stream->write_completed_queue, &pq);
+#if __GNUC__ >= 12
+#pragma GCC diagnostic pop
+#endif  // __GNUC__ >= 12
 
   while (!QUEUE_EMPTY(&pq)) {
     /* Pop a req off write_completed_queue. */
diff --git a/src/uv-common.c b/src/uv-common.c
index 3c65476a..fd56b013 100644
--- a/src/uv-common.c
+++ b/src/uv-common.c
@@ -444,7 +444,16 @@ void uv_walk(uv_loop_t* loop, uv_walk_cb walk_cb, void* arg) {
   QUEUE* q;
   uv_handle_t* h;
 
+// FIXME: GCC 12.1 gives a possibly real warning, but we don't know how to fix
+// it
+#if __GNUC__ >= 12
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdangling-pointer="
+#endif  // __GNUC__ >= 12
   QUEUE_MOVE(&loop->handle_queue, &queue);
+#if __GNUC__ >= 12
+#pragma GCC diagnostic pop
+#endif  // __GNUC__ >= 12
   while (!QUEUE_EMPTY(&queue)) {
     q = QUEUE_HEAD(&queue);
     h = QUEUE_DATA(q, uv_handle_t, handle_queue);
-- 
2.36.1

