From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: crueter <crueter@eden-emu.dev>
Date: Mon, 20 Jul 2026 23:08:48 -0400
Subject: [PATCH 2/3] Do not wrap cholmod in extern C

---
 src/mrcal_wrapper.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/mrcal_wrapper.h b/src/mrcal_wrapper.h
index 0a36d4702d59a06e81d15a139b27c2cb85c6e0e6..b71921e5d423f38ad5de138887d0fdac57816ab0 100644
--- a/src/mrcal_wrapper.h
+++ b/src/mrcal_wrapper.h
@@ -21,11 +21,8 @@ extern "C" {
 
 } // extern "C"
 
-// Seems like these people don't properly extern-c their headers either
-extern "C" {
 #include <suitesparse/SuiteSparse_config.h>
-#include <suitesparse/cholmod_core.h>
-} // extern "C"
+#include <suitesparse/cholmod.h>
 
 #include <memory>
 #include <opencv2/opencv.hpp>
