From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: crueter <crueter@eden-emu.dev>
Date: Mon, 20 Jul 2026 23:03:38 -0400
Subject: [PATCH 2/2] Don't wrap suitesparse in extern C

---
 src/mrcal_wrapper.h | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/mrcal_wrapper.h b/src/mrcal_wrapper.h
index 0a36d4702d59a06e81d15a139b27c2cb85c6e0e6..d9df64d412ec08522c4a7eb9df861436a3e46622 100644
--- a/src/mrcal_wrapper.h
+++ b/src/mrcal_wrapper.h
@@ -21,11 +21,8 @@ extern "C" {

 } // extern "C"

-// Seems like these people don't properly extern-c their headers either
-extern "C" {
 #include <suitesparse/SuiteSparse_config.h>
 #include <suitesparse/cholmod_core.h>
-} // extern "C"

 #include <memory>
 #include <opencv2/opencv.hpp>

