From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: crueter <crueter@eden-emu.dev>
Date: Tue, 21 Jul 2026 00:19:57 -0400
Subject: [PATCH 3/3] Fix compilation on OpenCV 5

---
 src/mrcal_wrapper.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/mrcal_wrapper.cpp b/src/mrcal_wrapper.cpp
index 3a9a5519e670b0c867281f1e05aa2046b938fa8d..fbd785ba263d8c61dc92010a824a7b528c5502a6 100644
--- a/src/mrcal_wrapper.cpp
+++ b/src/mrcal_wrapper.cpp
@@ -26,6 +26,11 @@
 #include <stdexcept>
 #include <vector>
 
+#include <opencv2/core/version.hpp>
+#if CV_VERSION_MAJOR >= 5
+#include <opencv2/geometry/3d.hpp>
+#endif
+
 using namespace cv;
 
 class CholmodCtx {
@@ -348,7 +353,7 @@ mrcal_pose_t getSeedPose(const mrcal_point3_t *c_observations_board_pool,
   }
 
   // Initial guess at intrinsics
-  Mat cameraMatrix = (Mat_<double>(3, 3) << fx, 0, cx, 0, fy, cy, 0, 0, 1);
+  Mat cameraMatrix{{3, 3}, {fx, 0.0, cx, 0.0, fy, cy, 0.0, 0.0, 1.0}};
   Mat distCoeffs = Mat(4, 1, CV_64FC1, Scalar(0));
 
   Mat_<double> rvec, tvec;
