From 61646f76602a77059c18298caa575b95b702c94c Mon Sep 17 00:00:00 2001
From: PJ Reiniger <pj.reiniger@gmail.com>
Date: Tue, 24 May 2022 01:24:31 -0400
Subject: [PATCH 3/3] Add advapi pragma

---
 Main/StackWalker/StackWalker.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Main/StackWalker/StackWalker.cpp b/Main/StackWalker/StackWalker.cpp
index 48c7c57..6f0fbf2 100644
--- a/Main/StackWalker/StackWalker.cpp
+++ b/Main/StackWalker/StackWalker.cpp
@@ -88,6 +88,7 @@
 #include <stdlib.h>
 #include <tchar.h>
 #pragma comment(lib, "version.lib") // for "VerQueryValue"
+#pragma comment(lib, "Advapi32.lib") // for "GetUserName"
 #pragma warning(disable : 4826)
 
 #ifdef UNICODE
